backend/app/resources.py

```diff
@@ -94,9 +103,86 @@ class SubcategoryList(Resource):
     def get(self):
         return marshal(Subcategory.query.all(), subcategory_marshaller)
 
+    @jwt_required
     def post(self):
         args = creating_subcategory_parser.parse_args()
-        subcategory = Repository.create_and_add(args)
+        subcategory = Repository.create_and_add(Subcategory, args)
         if not subcategory:
-            return 404
+            return {'message': 'Something went wrong'}, 500
         return marshal(subcategory, subcategory_marshaller), 201
+
+
+class UserRegistration(Resource):
+    def post(self):
+        args = user_parser.parse_args()
+
+        try:
+            new_user = User(username=args['username'])
+            db.session.add(new_user)
+            new_user.password_hash = args['password']
+            db.session.commit()
+            access_token = create_access_token(identity=args['username'])
+            refresh_token = create_refresh_token(identity=args['username'])
+            return {
+                       'message': 'User {} was created'.format(args['username']),
+                       'access_token': access_token,
+                       'refresh_token': refresh_token
+                   }, 201
+        except:
+            return {'message': 'Something went wrong'}, 500
+
+
+class UserLogin(Resource):
+    def post(self):
+        args = user_parser.parse_args()
+        current_user = User.query.filter_by(username=args['username']).first_or_404()
+        if current_user.check_password(args['password']):
+            access_token = create_access_token(identity=args['username'])
+            refresh_token = create_refresh_token(identity=args['username'])
+            return {
+                       'message': 'Logged in as {}'.format(current_user.username),
+                       'access_token': access_token,
+                       'refresh_token': refresh_token
+                   }, 201
+        else:
+            return {'message': 'Wrong credentials'}
+
+
+class UserLogoutAccess(Resource):
+    @jwt_required
+    def post(self):
+        jti = get_raw_jwt()['jti']
+        try:
+            revoked_token = RevokedTokenModel(jti=jti)
+            db.session.add(revoked_token)
```

Comment:
I believe that `db.session.add` and `commit` you have as `add_to_database` method. Why don't use it? :D